src/Algolia.Search/Clients/SearchIndex.cs

```diff
@@ -1061,5 +1062,25 @@ public ClearSynonymsResponse
                     $"/1/indexes/{_urlEncodedIndexName}/task/{taskId}", CallType.Read, requestOptions, ct)
                 .ConfigureAwait(false);
         }
+
+        /// <inheritdoc />
+        public bool Exists() => AsyncHelper.RunSync(() => ExistsAsync());
+
+        /// <inheritdoc />
+        public async Task<bool> ExistsAsync(CancellationToken ct = default)
+        {
+            try
+            {
+                await GetSettingsAsync(ct: ct);
+            }
+            catch (AlgoliaApiException ex)
+            {
+                if (ex.HttpErrorCode == 404)
```

Comment:
Since there's an `if` statement here, if we catch an `AlgoliaApiException` and it's not a 404, the method won't return nor throw, so we may want to throw it. If we do, then we may want to write a test with a mock on `GetSettingsAsync` that throws an `AlgoliaApiException` which isn't a 404. What do you think @Ant-hem ?

Reply:
Yes we are missing something there @sarahdayan . We have two solutions:

- 1st (The new way) -> replace `catch (AlgoliaApiException ex)` with `catch (AlgoliaApiException ex) when (ex.HttpErrorCode == 404)`. This syntax is underhood handling the `throw;` for other cases. More info about `catch() when()` [here](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/when).

- 2nd (The old way) -> We could add a `throw;` clause after the `if` block.

I would go for the 1st one.

Reply:
100% agree, solution 1 is much more elegant and it lets us rely on exception bubbling, hence we don't have to write an extra test. Will fix!

Reply:
Done :)